src/faction_camp.cpp

```diff
@@ -0,0 +1,3349 @@
+                if( comp != nullptr ) {
+                    popup( _( "%s returns from your garage..." ), comp->name.c_str() );
+                    camp_companion_return( *comp );
+                }
+                break;
+            }
+        }
+    }
+
+    g->draw_ter();
+    wrefresh( g->w_terrain );
+
+    return true;
+}
+
+// camp faction companion mission start functions
+void talk_function::start_camp_upgrade( npc &p, const std::string &bldg )
+{
+    std::vector<std::shared_ptr<npc>> npc_list = companion_list( p, "_faction_upgrade_camp" );
+    if( !npc_list.empty() ) {
+        popup( _( "You already have a companion upgrading the camp." ) );
+        return;
+    }
+    const recipe *making = &recipe_id( bldg ).obj();
+    //Stop upgrade if you don't have materials
+    inventory total_inv = g->u.crafting_inventory();
+    if( making->requirements().can_make_with_inventory( total_inv, 1 ) ) {
+        int need_food = time_to_food( time_duration::from_turns( making->time / 100 ) );
+        if( camp_food_supply() < need_food && bldg != "faction_base_camp_1" ) {
+            popup( _( "You don't have enough food stored to feed your companion." ) );
+            return;
+        }
+        npc *comp = individual_mission( p, _( "begins to upgrade the camp..." ), "_faction_upgrade_camp",
+                                        false, {},
+                                        making->skill_used.obj().ident().c_str(), making->difficulty );
+        if( comp != nullptr ) {
+            comp->companion_mission_time_ret = calendar::turn + time_duration::from_turns(
+                                                   making->time / 100 ) ;
+            g->u.consume_components_for_craft( making, 1, true );
+            g->u.invalidate_crafting_inventory();
+        }
+    } else {
+        popup( _( "You don't have the materials for the upgrade." ) );
+    }
+}
+
+void talk_function::start_cut_logs( npc &p )
+{
+    std::vector<std::shared_ptr<npc>> npc_list = companion_list( p, "_faction_camp_cut_log" );
+    if( !npc_list.empty() ) {
+        popup( _( "There are too many companions working on this mission!" ) );
+        return;
+    }
+    std::vector<std::string> log_sources = { "forest", "forest_thick", "forest_water" };
+    popup( _( "Forests and swamps are the only valid cutting locations." ) );
+    const tripoint omt_pos = p.global_omt_location();
+    tripoint forest = om_target_tile( omt_pos, 1, 50, log_sources );
+    if( forest != tripoint( -999, -999, -999 ) ) {
+        int tree_est = om_harvest_trees( p, forest, .50, false, false );
+        int tree_young_est = om_harvest_ter( p, forest, ter_id( "t_tree_young" ), .50, false );
+        int dist = rl_dist( forest.x, forest.y, omt_pos.x, omt_pos.y );
+        //Very roughly what the player does + 6 hours for prep, clean up, breaks
+        time_duration chop_time = 6_hours + ( 1_hours * tree_est ) + ( 7_minutes * tree_young_est );
+        //Generous to 
```

Comment:
Don't define variables at the top of functions like in C. Define them as late as possible. In this case below when you actually calculate its value. And than you can use `std::max` to ensure it's not zero: `const int trips = std::max( 1.0, ceil( total_m / max_m ) );`

Reply:
I actually cleaned this up a lot more in a separate commit when I started adding new functionality, but this is a fine time to fix this. So fixed.